apps/api/internal/api/v1/categories-aliases/service.go

```diff
@@ -0,0 +1,114 @@
+package categories_aliases
+
+import (
+	"net/http"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/samber/do"
+	"github.com/satont/tsuwari/apps/api/internal/di"
+	"github.com/satont/tsuwari/apps/api/internal/interfaces"
+	"github.com/satont/tsuwari/apps/api/internal/types"
+	model "github.com/satont/tsuwari/libs/gomodels"
+	uuid "github.com/satori/go.uuid"
+	"gorm.io/gorm"
+)
+
+func handleGetCategoryAliases(channelId string, services types.Services) ([]model.ChannelCategoryAlias, error) {
+	db := do.MustInvoke[*gorm.DB](di.Provider)
+
+	var aliases []model.ChannelCategoryAlias
+	err := db.Where(`"channelId" = ?`, channelId).Find(&aliases).Error
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, err.Error())
+	}
+
+	return aliases, nil
+}
+
+func handlePost(channelId string, dto *categoryAliasDto, services types.Services) (*model.ChannelCategoryAlias, error) {
+	logger := do.MustInvoke[interfaces.Logger](di.Provider)
+	db := do.MustInvoke[*gorm.DB](di.Provider)
+
+	existedAlias := &model.ChannelCategoryAlias{}
+	err := db.Where(`"channelId" = ? AND "alias" = ?`, channelId, dto.Alias).First(existedAlias).Error
+	if err == nil {
+		return nil, fiber.NewError(http.StatusBadRequest, "alias with this name already exists")
```

Comment:
Error can happend not only when alias not found, so you need to cover that case.

Reply:
Fixed, huge mistake by me.